### ISSUE TITLE
maybe add this testing effect: Sequence

### ISSUE THREAD

**Issue by radix** [radix/effect#7]
# Untested

```python
    class Sequence(object):
        """
        An Effect intent that returns a different result each time it's performed.
        """
        def __init__(self, responses):
            """
            :param list responses: Responses to return in sequence. Instances of
                exception will be raised.
            """
            self.responses = responses
            self._index = 0

        def perform_effect(self, dispatcher):
            """Return or raise the next result in sequence."""
            index = self._index
            self._index += 1
            response = self.responses[index]
            if isinstance(response, Exception):
                raise response
            else:
                return response
```

**Comment by cyli** [comment]
Is this task resolved with #45?

**Comment by radix** [comment]
Hmm.

Well, this is very different from #45, in that it's an intent, not a dispatcher, but I do think that SequenceDispatcher does basically satisfy the same goal. Given that I'll close this ticket, but anyone should feel free to reopen if there's a good use case.
